devlib/target.py

```diff
@@ -153,6 +155,7 @@ def __init__(self,
                  ):
         self.connection_settings = connection_settings or {}
         self.platform = platform or Platform()
+        self.connection_settings['platform'] = self.platform
```

Comment:
platform will not be passed as an argument into instantiation of any connection, not just gem5. So all existing connection implementations must be updated to (optionally) expect it.